app/controllers/contact_embed_controller.rb

```diff
@@ -7,24 +7,30 @@ class ContactEmbedController < ApplicationController
   end
 
   def new
+    @edition = Edition.find_current(document: params[:document])
     render layout: rendering_context
   end
 
   def create
     result = ContactEmbed::CreateInteractor.call(params: params)
-    markdown_code, issues = result.to_h.values_at(:markdown_code, :issues)
+
+    edition, markdown_code, issues = result.to_h.values_at(:edition,
```

Comment:
I'm not sure why `edition` needs to be returned by `ContactEmbed::CreateInteractor`. Is it possible to do the same thing as `new` and call `@edition = Edition.find_current(document: params[:document])`?

Reply:
`edition` is required because the govspeak snipped for contacts require an edition in order to rendered.

It may be possible, but by doing it this way we are consistent with how we do it for [images](https://github.com/alphagov/content-publisher/blob/master/app/controllers/images_controller.rb#L12) and [file attachments](https://github.com/alphagov/content-publisher/blob/master/app/controllers/file_attachments_controller.rb#L25) — not really my call to change this pattern.

Reply:
I had a look at those examples, and they lock the edition as well, which doesn't seem to be needed for inserting a contact?

Reply:
As far as I got it back then (3 months ago) we needed to lock it for the contacts too. I'll be sneaky and @benthorner on this — as he has more background on it than I do.